src/material/dialog/testing/dialog-harness.ts

```diff
@@ -10,13 +10,25 @@ import {ContentContainerComponentHarness, HarnessPredicate, TestKey} from '@angu
 import {DialogRole} from '@angular/material/dialog';
 import {DialogHarnessFilters} from './dialog-harness-filters';
 
+/** Selectors for different sections of the mat-dialog that can contain user content. */
+export const enum MatDialogSection {
+  TITLE = '.mat-dialog-title',
+  CONTENT = '.mat-dialog-content',
+  ACTIONS = '.mat-dialog-actions'
+}
+
 /** Harness for interacting with a standard `MatDialog` in tests. */
-export class MatDialogHarness extends ContentContainerComponentHarness<string> {
+export class MatDialogHarness extends ContentContainerComponentHarness<MatDialogSection | string> {
```

Comment:
Not sure if we can do anything about it, but I think that the `| string` here basically removes any type checking for `MatDialogSection`.

Reply:
Yeah I left that as a reminder to eventually change it. I added a @breaking-change comment as well